core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -195,6 +211,74 @@ void* HIPHostPinnedSpace::impl_allocate(
 
   return ptr;
 }
+
+void* HIPManagedSpace::allocate(const size_t arg_alloc_size) const {
+  return allocate("[unlabeled]", arg_alloc_size);
+}
+void* HIPManagedSpace::allocate(const char* arg_label,
+                                const size_t arg_alloc_size,
+                                const size_t arg_logical_size) const {
+  return impl_allocate(arg_label, arg_alloc_size, arg_logical_size);
+}
+void* HIPManagedSpace::impl_allocate(
+    const char* arg_label, const size_t arg_alloc_size,
+    const size_t arg_logical_size,
+    const Kokkos::Tools::SpaceHandle arg_handle) const {
+  void* ptr = nullptr;
+
+  if (arg_alloc_size > 0) {
+    if (is_first_hip_managed_allocation.exchange(false) &&
+        Kokkos::show_warnings()) {
+      if (!Kokkos::Impl::hip_driver_check_page_migration(m_device)) {
+        std::cerr << R"warning(
+Kokkos::HIP::allocation WARNING: The combination of device and system configuration
+                                 does not support page migration between device and host.
+                                 HIPManagedSpace might not work as expected.
+                                 Please refer to the Rocm documentation on unified/managed memory.
+               )warning"
+                  << std::endl;
+      }
+
+      // check for correct runtime environment
+      const char* hsa_xnack = std::getenv("HSA_XNACK");
+      if (!hsa_xnack)
+        std::cerr << R"warning(
+Kokkos::HIP::runtime WARNING: Kokkos did not find an environment variable 'HSA_XNACK'
+                              for the current process.
+                              Nevertheless, xnack is enabled for all processes if
+                              amdgpu.noretry=0 was set in the Linux kernel boot line.
+                              Without xnack enabled, Kokkos::HIPManaged might not behave
+                              as expected.)warning"
+                  << std::endl;
+      else if (Kokkos::Impl::strcmp(hsa_xnack, "1"))
```

Comment:
```suggestion
      else if (Kokkos::Impl::strcmp(hsa_xnack, "1") != 0)
```

Reply:
I see that it adds readability here, even though I initially thought that people would know strcmp